HdProduction.HelpDesk.Domain/Safeguards/TicketPrioritySafeguard.cs

```diff
@@ -0,0 +1,32 @@
+using System.Threading.Tasks;
+using HdProduction.HelpDesk.Domain.Contract;
+using HdProduction.HelpDesk.Domain.Exceptions;
+
+namespace HdProduction.HelpDesk.Domain.Safeguards
+{
+    public class TicketPrioritySafeguard : ITicketPrioritySafeguard
+    {
+        private readonly ITicketPriorityRepository _ticketPriorityRepository;
+
+        public TicketPrioritySafeguard(ITicketPriorityRepository ticketPriorityRepository)
+        {
+            _ticketPriorityRepository = ticketPriorityRepository;
+        }
+
+        public async Task EnsureNameAsync(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new BusinessLogicException("Name can't be empty");
+            }
+            if (name.Length > 32)
+            {
+                throw new BusinessLogicException("Name is too long");
+            }
+            if (await _ticketPriorityRepository.FindByNameAsync(name) != null)
+            {
+                throw new BusinessLogicException("There can not be duplicated names");
```

Comment:
We have to determine a place for such literals with exceptions name. Do you have any thoughts about it?

Reply:
I've created ExceptionsHelper for this, look pls

Reply:
Helper resolves another problem. A little bit easier to search or modify when messages moved to constants instead of its direct using.